projects/angular/src/utils/dom-adapter/dom-adapter.ts

```diff
@@ -14,10 +14,16 @@ import { Injectable } from '@angular/core';
 
 @Injectable()
 export class DomAdapter {
+  /* 
+    We clone the element and take its measurements from outside the grid
+    so we don't trigger reflow for the whole datagrid.
+  */
   userDefinedWidth(element: HTMLElement): number {
-    element.classList.add('datagrid-cell-width-zero');
-    const userDefinedWidth = this.clientRect(element).width;
-    element.classList.remove('datagrid-cell-width-zero');
+    const clonedElement = element.cloneNode(true) as HTMLElement;
+    clonedElement.classList.add('datagrid-cell-width-zero');
+    document.body.appendChild(clonedElement);
```

Comment:
Pitfall: https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode
`CloneNode` possible `id` duplication, possibly the `name` attributes as well.
Change the cloned node ID to something else. Add `-clone` at the end of it.
```suggestion
    clonedElement.id = clonedElement.id + '-clone';
    document.body.appendChild(clonedElement);
```

Reply:
We are removing the clone instantly so shouldn't be a problem

Reply:
I pointed it out to be 100% sure in uniqueness of the `id`. We don't know how datagrid may be used and is low effort to update the `id`, unless it leads to some performance degradation.
There might be very slim corner case that might lead to some unexpected results from not changing the `id`.